### PR TITLE
Bugfix/parse dates

### PR DESCRIPTION
**Pull request by wrp801**
Updates: 
- Changed upload method to use gzip'd csv's instead of parquet files
- Added additional logic for properly parsing dates when detected

### CODE REVIEW

File: snowflake_blueprints/upload_file.py [hunk 1]
```diff
@@ -106,7 +107,12 @@ def map_snowflake_to_pandas(snowflake_data_types):
         field = item[0]
         dtype = item[1]
         try:
-            converted = snowflake_to_pandas.get(str(dtype).upper())
+            # converted = snowflake_to_pandas.get(str(dtype).upper())
```
Comment (forstmeier) [inline snowflake_blueprints/upload_file.py]: Not needed?

File: snowflake_blueprints/upload_file.py [hunk 2]
```diff
@@ -116,13 +122,26 @@ def map_snowflake_to_pandas(snowflake_data_types):
     return pandas_dtypes
 
 
+def get_pandas_dates(pandas_datatypes: dict) -> list:
```
Comment (forstmeier) [inline snowflake_blueprints/upload_file.py]: Aren't there two values being returned? I'm not sure how that's represented in type hinting.

File: snowflake_blueprints/upload_file.py [hunk 3]
```diff
@@ -193,11 +222,57 @@ def convert_to_parquet(source_full_path, table_name, snowflake_datatypes):
...
+
 def execute_put_command(db_connection, file_path, table_name, results_dict):
     """
     Execute the PUT command against Snowflake and store the results.
     """
+    # NOTE testing creating a staging area
+    # create_sql = f"""CREATE TEMPORARY STAGE stg FILE_FORMAT = (TYPE = 'PARQUET'); """
+    # db_connection.execute(create_sql)  # create the staging table
     put_statement = f"PUT file://{file_path}/part.*.parquet '@%\"{table_name}\"'"
+    # put_statement = f"PUT file://{file_path}/part.*.parquet @stg;"
```
Comment (forstmeier) [inline snowflake_blueprints/upload_file.py]: Remove unused comments.

File: snowflake_blueprints/upload_file.py [hunk 4]
```diff
@@ -193,11 +222,57 @@ def convert_to_parquet(source_full_path, table_name, snowflake_datatypes):
...
+    datatypes = map_snowflake_to_pandas(snowflake_datatypes)
+    dates = None
+    pandas_datatypes = None
+    if datatypes is not None:
+        dates, pandas_datatypes = get_pandas_dates(datatypes)
+    df = dd.read_csv(source_full_path, dtype=pandas_datatypes, parse_dates=dates,
+                     date_parser=lambda x: pd.to_datetime(x).to_datetime64())
+    df.columns = map(lambda x: str(x).upper(), df.columns)
+    df.to_csv(full_path, compression='gzip', index=False)
+    # return csv_path
```
Comment (forstmeier) [inline snowflake_blueprints/upload_file.py]: Remove.

File: snowflake_blueprints/upload_file.py [hunk 5]
```diff
@@ -193,11 +222,57 @@ def convert_to_parquet(source_full_path, table_name, snowflake_datatypes):
...
+    return full_path
+
+
+def put_csv(db_connection, file_path, table_name, results_dict):
+    """
+    Execute the PUT command against Snowflake and store the results.
+    """
+    put_statement = f"PUT file://{file_path} '@%\"{table_name}\"'"
+    # put_statement = f"PUT file://{file_path}/part.*.parquet @stg;"
```
Comment (forstmeier) [inline snowflake_blueprints/upload_file.py]: Remove.

**Review by forstmeier — CHANGES_REQUESTED** [review]
Minor cleanup otherwise I think it's fine.

File: snowflake_blueprints/upload_file.py (hunk 1 above)
Comment (wrp801) [reply snowflake_blueprints/upload_file.py]: Done

File: snowflake_blueprints/upload_file.py (hunk 2 above)
Comment (wrp801) [reply snowflake_blueprints/upload_file.py]: Done

File: snowflake_blueprints/upload_file.py (hunk 3 above)
Comment (wrp801) [reply snowflake_blueprints/upload_file.py]: Done

File: snowflake_blueprints/upload_file.py (hunk 4 above)
Comment (wrp801) [reply snowflake_blueprints/upload_file.py]: Done

File: snowflake_blueprints/upload_file.py (hunk 5 above)
Comment (wrp801) [reply snowflake_blueprints/upload_file.py]: Done

**Review by forstmeier — APPROVED** [review]

**Pull request merged**